mlflow/server/js/src/experiment-tracking/components/MetricsPlotView.test.js

```diff
@@ -105,19 +235,43 @@ describe('unit tests', () => {
       ...minimalPropsForLineChart,
       lineSmoothness: 50,
     };
+    minimalPropsForLineChartWithNaNs = {
+      ...minimalPropsForLineChart,
+      metrics: metricsForLineWithNaNs,
+      metricKeys: metricsForLineWithNaNs.map((metric) => metric.metricKey),
+    };
+    minimalPropsForSmoothedLineChartWithNaNs = {
+      ...minimalPropsForLineChartWithNaNs,
+      lineSmoothness: 50,
+    };
     minimalPropsForBarChart = {
       ...minimalPropsForLineChart,
       metrics: metricsForBar,
-      metricKeys: ['metric_0'],
+      metricKeys: metricsForBar.map((metric) => metric.metricKey),
       chartType: CHART_TYPE_BAR,
     };
+    minimalPropsForBarChartWithNaNs = {
+      ...minimalPropsForLineChart,
+      metrics: metricsForBarWithNaNs,
+      metricKeys: metricsForBarWithNaNs.map((metric) => metric.metricKey),
+    };
   });
 
   test('should render with minimal props without exploding', () => {
     wrapper = shallow(<MetricsPlotView {...minimalPropsForLineChart} />);
     expect(wrapper.length).toBe(1);
   });
 
+  test('should render bar chart successfully for metrics containing NaN values', () => {
+    wrapper = shallow(<MetricsPlotView {...minimalPropsForBarChartWithNaNs} />);
+    expect(wrapper.length).toBe(1);
+  });
+
+  test('should render line chart successfully for metrics containing NaN values', () => {
```

Comment:
Do you think we need one of these tests for `minimalPropsForSmoothedLineChartWithNaNs`?

Reply:
Great question! This is tested here: https://github.com/mlflow/mlflow/blob/add53130b009dc42f92cddfbb78dacef6c288ce1/mlflow/server/js/src/experiment-tracking/components/MetricsPlotView.test.js#L400-L401